### PR TITLE
Add test for saving encrypted tox data to a file.

### PR DESCRIPTION
**Pull request by pranomostro**
tox_get_savedata(t, clear);

writes no data.
This seems to be the real issue here, if I'm not mistaken.

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/298)

### CODE REVIEW

**Comment by zetok** [comment]
Could you add license header?

**Comment by iphydf** [comment]
Thanks, can you run `other/astyle/format-source`? Or copy/paste the patch from the failed travis build and apply it.

File: auto_tests/file_saving_test.c [hunk 1]
```diff
@@ -0,0 +1,133 @@
+/* file_saving_test.c
+ *
+ * Small test for checking if obtaining savedata, saving it to disk and using
+ * works correctly.
+ *
+ *  Copyright (C) 2013 Tox project All Rights Reserved.
```
Comment (iphydf) [inline auto_tests/file_saving_test.c]: 2016

File: auto_tests/file_saving_test.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <tox/tox.h>
+#include <tox/toxencryptsave.h>
+
+char* pphrase="bar", * name="foo";
```
Comment (iphydf) [inline auto_tests/file_saving_test.c]: `static` and `const`

File: auto_tests/file_saving_test.c [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	off_t sz;
+	int fd;
+	TOX_ERR_ENCRYPTION eerr;
+
+	tox_options_default(&to);
+	t=tox_new(&to, NULL);
+
+	tox_self_set_name(t, (uint8_t*)name, strlen((char*)name), NULL);
+
+	fd = open("save", O_WRONLY | O_TRUNC | O_CREAT , 0666);
```
Comment (iphydf) [inline auto_tests/file_saving_test.c]: Prefer C library functions to POSIX ones. This should be `fopen`.

**Comment by iphydf** [comment]
Also, hook it up to the cmake build. Look for auto_test in CMakeLists.txt. This way, it will run on travis, so we can see it fail (and pass, in the future).

File: auto_tests/file_saving_test.c (hunk 1 above)
Comment (pranomostro) [inline auto_tests/file_saving_test.c]: done.

File: auto_tests/file_saving_test.c (hunk 2 above)
Comment (pranomostro) [inline auto_tests/file_saving_test.c]: done.

File: auto_tests/file_saving_test.c (hunk 3 above)
Comment (pranomostro) [inline auto_tests/file_saving_test.c]: done.

File: auto_tests/file_saving_test.c [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+ *  You should have received a copy of the GNU General Public License
+ *  along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <tox/tox.h>
```
Comment (iphydf) [inline auto_tests/file_saving_test.c]: `../toxcore/tox.h` and `../toxencryptsave/toxencryptsave.h`

File: auto_tests/file_saving_test.c [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+#include <tox/toxencryptsave.h>
+
+static const char *pphrase = "bar", * name = "foo";
```
Comment (iphydf) [inline auto_tests/file_saving_test.c]: Can you make two separate declarations or change it from `* name` to `*name`?

File: auto_tests/file_saving_test.c [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    free(clear);
+    fclose(f);
+    tox_kill(t);
+}
+
+int main(void)
+{
+    tse();
+    tsd();
+
```
Comment (iphydf) [inline auto_tests/file_saving_test.c]: Delete the temporary file.

**Comment by robinlinden** [comment]
Reviewed 1 of 2 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[auto_tests/file_saving_test.c, line 51 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/298#-KYJIo5DERsGP8kfyt8E:-KYJIo5E-7AMzL2uNgWY:bby1vrz) ([raw file](https://github.com/toktok/c-toxcore/blob/28fed777f0f1d11c637de900903934e41c30c629/auto_tests/file_saving_test.c#L51)):*
> ```C
> 
>     sz = tox_get_savedata_size(t);
>     clear = malloc(sz);
> ```

`(uint8_t *)malloc`

---

*[auto_tests/file_saving_test.c, line 57 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/298#-KYJIhm-76Eosf4dgFkx:-KYJIhm00lFVLbbJ2k05:bby1vrz) ([raw file](https://github.com/toktok/c-toxcore/blob/28fed777f0f1d11c637de900903934e41c30c629/auto_tests/file_saving_test.c#L57)):*
> ```C
> 
>     sz += TOX_PASS_ENCRYPTION_EXTRA_LENGTH;
>     cipher = malloc(sz);
> ```

`(uint8_t *)malloc`

---

*[auto_tests/file_saving_test.c, line 76 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/298#-KYJJCxU9hoFqjU1GYrf:-KYJJCxUDV_ajGc927X-:b-qy1wvu) ([raw file](https://github.com/toktok/c-toxcore/blob/28fed777f0f1d11c637de900903934e41c30c629/auto_tests/file_saving_test.c#L76)):*
> ```C
> {
>     uint8_t readname[TOX_MAX_NAME_LENGTH];
>     uint8_t *clear, * cipher;
> ```

`*cipher`

---

*[auto_tests/file_saving_test.c, line 88 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/298#-KYJJ9f8CeUWUQcgbFUz:-KYJJ9f8DFr9YB7FnUo5:bby1vrz) ([raw file](https://github.com/toktok/c-toxcore/blob/28fed777f0f1d11c637de900903934e41c30c629/auto_tests/file_saving_test.c#L88)):*
> ```C
>     fseek(f, 0, SEEK_SET);
> 
>     cipher = malloc(sz);
> ```

`(uint8_t *)malloc`

---

*[auto_tests/file_saving_test.c, line 89 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/298#-KYJJA_47_ayHH7QhdOI:-KYJJA_40YlTFjvYTsYj:bby1vrz) ([raw file](https://github.com/toktok/c-toxcore/blob/28fed777f0f1d11c637de900903934e41c30c629/auto_tests/file_saving_test.c#L89)):*
> ```C
> 
>     cipher = malloc(sz);
>     clear = malloc(sz - TOX_PASS_ENCRYPTION_EXTRA_LENGTH);
> ```

`(uint8_t *)malloc`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/298)*

**Comment by iphydf** [comment]
I've made a PR to your branch. Please take a look and incorporate the changes if you agree with them.

**Comment by iphydf** [comment]
See https://github.com/pranomostro/c-toxcore/pull/2. Next steps: merge that PR, squash all commits into one, rebase on master, and then I'll merge it.

**Pull request closed without merging**